### DISCUSSION TITLE
webERP v4.15.2: Notice: Undefined variable: Time in weberp/includes/DateFunctions.inc on line 345

### DISCUSSION THREAD

**Discussion by pakricard** [timschofield/webERP#191] [Problems / Bugs] [answered]
I am trying to bring my heavily customized v4.13.1 to the current v4.15..2, before jumping to the new v5.

I get this: 
Notice Undefined variable: Time in weberp/includes/DateFunctions.inc on line 345

$Time gets a value on line 340 as:
```
	if (mb_strlen($Date_Array[2])>4) {
		$Time = mb_substr($Date_Array[2],3,8);
		$Date_Array[2]= mb_substr($Date_Array[2],0,2);
	}
```
but I guess there is missing an else structure defining $Time for the other cases. 

It's outside my circle of confidence to properly fix it. Any ideas how?

**Comment by timschofield** [comment, marked as answer]
Thanks very much Ricard. You are right, and I have uploaded a fix for this. I changed the code block to read:
	if (mb_strlen($Date_Array[2])>4) {
		$Time = mb_substr($Date_Array[2],3,8);
		$Date_Array[2]= mb_substr($Date_Array[2],0,2);
	} else {
		$Time = '0:0:0';
	}

Thanks for the report
Tim

**Reply by pakricard** [reply]
Tim: I pushed this change into my installation, and now there is no notice warning (thanks!), but I think line 343:
` 		$Time = '0:0:0';`
should read 
`		$Time = '00:00:00';`
at least in my locale. Should we have this “zero” time locale dependant? 
Function ConvertSQLDateTime should convert date and time from SQL format, and this bit is only for error catching in case the parameter is a date with no time in it. Maybe it is not relevant, though.

**Comment by timschofield** [comment]
Thanks Ricard. I have updated this. The correct string is as you say
"00:00:00". To be honest I have no idea how PHP would react to the single
zeros but we should be accurate anyway.

Thanks for this report,
Tim

On Tue, 12 Nov 2024 at 01:48, pakricard ***@***.***> wrote:

> Tim: I pushed this change into my installation, and now there is no notice
> warning (thanks!), but I think line 343:
> $Time = '0:0:0';
> should read
> $Time = '00:00:00';
> at least in my locale. Should we have this “zero” time locale dependant?
> Function ConvertSQLDateTime should convert date and time from SQL format,
> and this bit is only for error catching in case the parameter is a date
> with no time in it. Maybe it is not relevant, though.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/191#discussioncomment-11220396>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LG4LRNMHIRYEBVTAAL2AFM7DAVCNFSM6AAAAABQOLI6FWVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMRSGAZTSNQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
www.weberp.org <http://www.weberpafrica.com>
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Reply by dalers** [reply]
I suspect consumers of $Time (and perhaps unknowingly) either change string 00 to numeric 0 in a register, or pads the string on the left to convert string 0 to 00...
